tic-tac-toe.js

```diff
@@ -1,7 +1,77 @@
-function TicTacToe() {
+var TicTacToe = function() {
+ // initialize things when a new game starts
+ this.board = ["one", "two", "three", "four", "five", "six", "seven", "eight", "nine"];
+ this.player1 = true;
+ this.player2 = false;
+ this.score = [[9, 9, 9], [9, 9, 9], [9, 9, 9]];
+};
 
-}
 
-TicTacToe.prototype = {
-  
-}
+  TicTacToe.prototype.play = function() {
+//player 1 will be 1s and player 2 will be 2s.
+    var self = this;
+      self.board.forEach(function(cell) {
+        $("#" + cell).click(function() {
+          $("#" + cell).html(self.markBoard()); self.scoreGame($(this)); self.changePlayer();
+        });
+      }); //ends forEach
+    };  // ends play function
+
+//This function is called in .play
+  TicTacToe.prototype.markBoard = function() {
+    if (this.player1) {
+      return "<h1 class='x'>X</h1>";
+    } else {
+      return "<h1 class='o'>O</h1>";
+    }
+  };
+
+//This function is called in .play
+  TicTacToe.prototype.changePlayer = function() {
+    if (this.player1) {
+      this.player1 = false;
+      this.player2 = true;
+    } else {
+      this.player2 = false;
+      this.player1 = true;
+    }
+  };
+
+
+// This function is called in .play (before changePlayer)
+  TicTacToe.prototype.scoreGame = function(clicked) {
+// assign point depending on whose turn it is.
+    var point;
+    if (this.player1) {
+      point = 1;
+    } else {
+      point = 2;
+    }
+// put point in 2D scoring array
+    if (clicked.attr("id") === "one") {
```

Comment:
There is a way to have the clicked element contain information about the row and column in which it is associated and that would clean up this code. Let's say you have a class name like `r0c0` or `r0c1`. You can then retrieve the `0` and `1` accordingly and use that to update the `score` array.